server/src/main/java/org/elasticsearch/rest/action/search/RestMultiSearchAction.java

```diff
@@ -88,6 +88,14 @@ public String getName() {
     @Override
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
         MultiSearchRequest multiSearchRequest = parseRequest(request, allowExplicitIndex);
+
+        // Emit a single deprecation message if any search request contains types.
+        for (SearchRequest searchRequest : multiSearchRequest.requests()) {
+            if (searchRequest.types().length > 0) {
+                deprecationLogger.deprecatedAndMaybeLog("msearch_with_types", TYPES_DEPRECATION_MESSAGE);
+                break;
+            }
+        }
```

Comment:
Same here, is there a unit test already? Otherwise is is possible to add one?

Reply:
Likewise these deprecations are covered by `RestMultiSearchActionTests`.